GoogleUtilities/Logger/GULASLLogger.m

```diff
@@ -66,7 +66,7 @@ - (void)initializeLogger {
 - (void)dealloc {
 #pragma clang diagnostic push
 #pragma clang diagnostic ignored "-Wdeprecated-declarations"  // asl is deprecated
-  asl_release(self.aslClient);
+  asl_free(self.aslClient);
```

Comment:
Would it make sense to have availability check and use modern API if available, like:
```
if (@available(iOS 8.0, *)) {
    asl_release(self.aslClient);
  } else {
    asl_free(self.aslClient)
  }
```

Reply:
the `@available` built-in is only available on Xcode 9 and later, but we still support Xcode 8.

Reply:
Ideally yes, but we can't use `@available` with GoogleUtilities since Firestore has Xcode 8 compatibility tests run on Travis and `@available` only works with Xcode 9+ 😞

Reply:
Note that the pre-Xcode 9-requirements are only for Firestore and its dependents. The pre-iOS 8 - requirements are only for Analytics and its dependents.

Reply:
And `GoogleUtilities` is the very root dependency of Firebase.

Reply:
Makes sense, thank you for explanation.